internal/service/opensearchserverless/security_policy_data_source.go

```diff
@@ -0,0 +1,72 @@
+package opensearchserverless
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs/sdkdiag"
+)
+
+// @SDKDataSource("aws_opensearchserverless_security_policy")
+func DataSourceSecurityPolicy() *schema.Resource {
+	return &schema.Resource{
+		ReadWithoutTimeout: dataSourceSecurityPolicyRead,
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"policy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"policy_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
```

Comment:
[The API](https://docs.aws.amazon.com/opensearch-service/latest/ServerlessAPIReference/API_GetSecurityPolicy.html#API_GetSecurityPolicy_RequestSyntax) has two allowed values (`encryption`  or `network`), and those values are [defined in the AWS SDK](https://github.com/aws/aws-sdk-go-v2/blob/2b71c0eac6c4f32a20074c9f60b5ad5f93dff490/service/opensearchserverless/types/enums.go#L85C2-L85C2). Can you please add validation as such?

You can see an example here: 
https://github.com/hashicorp/terraform-provider-aws/blob/8f461fec79149dc1031ee843250ead0da9030483/internal/service/opensearchserverless/security_policy.go#L86C12-L86C12